### PR TITLE
Make backend.Workspaces accept a context

### PR DESCRIPTION
**Pull request by marcinwyszynski**
Given that many state backend methods use contexts in their calls, we will make the related interfaces accept context as an input to all methods. Since it's a ton of work, we go one by one. This time it's a method (Workspaces) where the context is fairly widely used.

related to https://github.com/opentofu/opentofu/issues/753

## Target Release

1.6.0

## Draft CHANGELOG entry

n/a, it's a purely internal change

### CODE REVIEW

File: internal/command/workspace_delete.go [hunk 1]
```diff
@@ -75,7 +75,9 @@ func (c *WorkspaceDeleteCommand) Run(args []string) int {
 	// This command will not write state
 	c.ignoreRemoteVersionConflict(b)
 
-	workspaces, err := b.Workspaces()
+	ctx := context.Background()
```
Comment (mbialon) [inline internal/command/workspace_delete.go:78]: Why not use `context.TODO()`?

File: internal/command/workspace_list.go [hunk 2]
```diff
@@ -58,7 +59,9 @@ func (c *WorkspaceListCommand) Run(args []string) int {
 	// This command will not write state
 	c.ignoreRemoteVersionConflict(b)
 
-	states, err := b.Workspaces()
+	ctx := context.Background()
```
Comment (mbialon) [inline internal/command/workspace_list.go:62]: Why not use `context.TODO()`?

File: internal/command/workspace_new.go [hunk 3]
```diff
@@ -90,7 +90,9 @@ func (c *WorkspaceNewCommand) Run(args []string) int {
 	// This command will not write state
 	c.ignoreRemoteVersionConflict(b)
 
-	workspaces, err := b.Workspaces()
+	ctx := context.Background()
```
Comment (mbialon) [inline internal/command/workspace_new.go:93]: Why not use `context.TODO()`?

File: internal/command/workspace_select.go [hunk 4]
```diff
@@ -83,7 +83,9 @@ func (c *WorkspaceSelectCommand) Run(args []string) int {
 		return 1
 	}
 
-	states, err := b.Workspaces()
+	ctx := context.Background()
```
Comment (mbialon) [inline internal/command/workspace_select.go:86]: Why not use `context.TODO()`?

File: internal/command/workspace_delete.go (hunk 1 above)
Comment (marcinwyszynski) [reply internal/command/workspace_delete.go]: Why not indeed. Fixed.

File: internal/command/workspace_new.go (hunk 3 above)
Comment (marcinwyszynski) [reply internal/command/workspace_new.go]: Same.

File: internal/command/workspace_list.go (hunk 2 above)
Comment (marcinwyszynski) [reply internal/command/workspace_list.go]: Fixed.

**Review by mbialon — APPROVED** [review]

**Pull request merged**